### ISSUE TITLE
Huge Autoencoder Scenario

### ISSUE THREAD

**Issue by Ahmadrezauf** [OATML/EVE#8]
Hello,
I was trying to use EVE on the BLAT_ECOLX dataset from your paper, using a huge architecture and a large latent dimension and a small learning rate, while setting the KL coefficient to zero. In an ideal scenario, the corresponding auto encoder should memorize every datapoint and result in a 0 BCE. However, the BCE doesn't change much in comparison to a small auto encoder and doesn't get close to zero but a high number (around 800). 
As you have reported your results on this dataset as well, I was wondering if there is any explanation for this phenomena.

**Comment by pascalnotin** [comment]
Hi @Ahmadrezauf,
The hypothesis that "BCE on the *training data* converges to 0 for a large network" should be true for a deterministic autoencoder. However, there are two sources of regularization in a bayesian VAE that should prevent that from happening: 1) sampling from the approximate posterior `z~q(z|x)` (true for any VAE) 2) sampling from decoder parameters (true for *bayesian* VAEs). Did you set the KL coefficient to zero for both? How small was your learning rate (it should not be too small either to allow convergence)?

**Comment by Ahmadrezauf** [comment]
Hi @pascalnotin Thanks a lot for your response. So I have set the KL to zero for both, and a very small learning rate such as 1e-5. The problem gets solved when removing the logsoftmax from the end of the decoder though. I don't understand that in the end as well, since the BCEwithLogits already applies a sigmoid to the create logits. With having that and a very small learning rate, the BCE doesn't go to zero.

**Comment by pascalnotin** [comment]
Interesting! Your response just reminded me that there is another reason why the BCE may not converge to zero (and why we use the BCEwithLogits): the vocabulary we use for modeling is comprised of the 20 standard amino acids. However, the sequences in the MSA that we use as our training data also contain gaps (dashes in the data), which we did no want to include in our output (we were mostly interested in predicting missense mutations; gaps in the MSA may have various explanations: indels, the sequence is only overlapping with a sub-domain, etc.). So whenever there is a position with a gap in the input, the resulting BCE will be non-zero regardless of the size of the VAE.
